source/binding/AssignmentExpressions.cpp

```diff
@@ -604,6 +605,11 @@ ConstantValue ConversionExpression::convert(EvalContext& context, const Type& fr
         return std::move(value);
     }
 
+    if (to.isUnpackedArrayOfByte()) {
+        const FixedSizeUnpackedArrayType* ct = &to.as<FixedSizeUnpackedArrayType>();
```

Comment:
When doing as<> casting on types you need to be careful to get the canonical type first, in case there are typedefs involved.

Reply:
Yes, I will change the code into
```c++
    if (to.isByteArray()) {
        const auto& ct = to.getCanonicalType();
        if (ct.hasFixedRange()) {
            const FixedSizeUnpackedArrayType* cf = &ct.as<FixedSizeUnpackedArrayType>();
            return value.convertToByteArray(cf->range.width(),
                                            to.getArrayElementType()->isSigned());
        }
    }
```